# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/rspamd/no-hyperscan.patch
# Copyright (C) 2023 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

From d907a95ac2e2cad6f7f65c4323f031f7931ae18b Mon Sep 17 00:00:00 2001
From: Vsevolod Stakhov <vsevolod@rspamd.com>
Date: Wed, 15 Nov 2023 13:43:05 +0000
Subject: [PATCH] [Minor] Fix build with no hyperscan

Issue: #4702
---
 src/libserver/rspamd_control.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/libserver/rspamd_control.c b/src/libserver/rspamd_control.c
index 986a4a2ea..69af0597f 100644
--- a/src/libserver/rspamd_control.c
+++ b/src/libserver/rspamd_control.c
@@ -922,6 +922,7 @@ rspamd_srv_handler(EV_P_ ev_io *w, int revents)
 				}
 				break;
 			case RSPAMD_SRV_HYPERSCAN_LOADED:
+#ifdef WITH_HYPERSCAN
 				/* Load RE cache to provide it for new forks */
 				if (rspamd_re_cache_is_hs_loaded(rspamd_main->cfg->re_cache) != RSPAMD_HYPERSCAN_LOADED_FULL ||
 					cmd.cmd.hs_loaded.forced) {
@@ -932,7 +933,9 @@ rspamd_srv_handler(EV_P_ ev_io *w, int revents)
 				}
 
 				/* After getting this notice, we can clean up old hyperscan files */
+
 				rspamd_hyperscan_notice_loaded();
+
 				msg_info_main("received hyperscan cache loaded from %s",
 							  cmd.cmd.hs_loaded.cache_dir);
 
@@ -945,6 +948,7 @@ rspamd_srv_handler(EV_P_ ev_io *w, int revents)
 				wcmd.cmd.hs_loaded.forced = cmd.cmd.hs_loaded.forced;
 				rspamd_control_broadcast_cmd(rspamd_main, &wcmd, rfd,
 											 rspamd_control_ignore_io_handler, NULL, worker->pid);
+#endif
 				break;
 			case RSPAMD_SRV_MONITORED_CHANGE:
 				/* Broadcast command to all workers */
